pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -199,6 +223,11 @@ def __init__(
 
         self._standard_gateset = gate_set >= {OpType.X, OpType.SX, OpType.Rz, OpType.CX}
 
+        self._primitive_gates = (
+            GateSet.X_SX_RZ_CX.value
+            if gate_set >= {OpType.X, OpType.SX, OpType.Rz, OpType.CX}
+            else GateSet.X_SX_RZ_ECR.value
```

Comment:
I'm not really happy with the syntax of this. 

If there's any suggestion for imporvement let me know